app/src/main/java/org/wikipedia/edit/EditSectionActivity.kt

```diff
@@ -712,12 +802,30 @@ class EditSectionActivity : BaseActivity(), ThemeChooserDialog.Callback {
         }
         binding.viewEditSectionError.isVisible = false
         if (editSummaryFragment.handleBackPressed()) {
-            supportActionBar?.title = getString(R.string.preview_edit_title)
+            ImageRecommendationsEvent.logAction("back", "editsummary_dialog", ImageRecommendationsEvent.getActionDataString(
+                filename = addImageTitle?.prefixedText.orEmpty(), recommendationSource = addImageSource.orEmpty(), acceptanceState = "accepted",
+                captionAdd = !intent.getStringExtra(InsertMediaActivity.RESULT_IMAGE_CAPTION).isNullOrEmpty(),
+                altTextAdd = !intent.getStringExtra(InsertMediaActivity.RESULT_IMAGE_ALT).isNullOrEmpty()), pageTitle.wikiSite.languageCode)
+            supportActionBar?.title = getString(R.string.edit_preview)
             return
         }
         if (editPreviewFragment.isActive) {
+            ImageRecommendationsEvent.logAction("back", "caption_preview", ImageRecommendationsEvent.getActionDataString(
+                filename = addImageTitle?.prefixedText.orEmpty(), recommendationSource = addImageSource.orEmpty(), acceptanceState = "accepted",
+                captionAdd = !intent.getStringExtra(InsertMediaActivity.RESULT_IMAGE_CAPTION).isNullOrEmpty(),
+                altTextAdd = !intent.getStringExtra(InsertMediaActivity.RESULT_IMAGE_ALT).isNullOrEmpty()), pageTitle.wikiSite.languageCode)
             editPreviewFragment.hide(binding.editSectionContainer)
             supportActionBar?.title = null
+
+            // If we came from the Image Recommendations workflow, bring back the Add Image activity.
+            if (invokeSource == Constants.InvokeSource.EDIT_ADD_IMAGE) {
+                // ...and reset the wikitext to the original, since the Add Image flow will re-
+                // modify it when the user returns to it.
+                sectionWikitext = sectionWikitextOriginal
+                binding.editSectionText.setText(sectionWikitext)
+
+                startInsertImageFlow()
```

Comment:
Not sure if there is a way of optimizing it when:

"Add image details" -> "Preview", and go back to the "Add image details".

Before going back to the "Add image details" screen, the wikitext editor screen (`EditSectionActivity`) will show up a bit, which may slightly confuse users.

Reply:
Yes, good observation - I was aware of this too, but not sure if there's any way to optimize it away. Unless you have an idea, it might need to be something we follow up on afterwards.

Reply:
Got it. I think it would be nice if we could change the loading behavior to a full-screen `ProgressBar` in `EditSectionActivity`, but it is sure that it will need some design feedback though.